### ISSUE TITLE
Content of annotation is missing lang attributes

### ISSUE THREAD

**Issue by gautierchomel** [edrlab/thorium-reader#2926]
The annotation navigation panel and the exported annotation do not contain information about the language of the contents.
Neither for the quoted text nor for the annotation value. 

Example of annotation navigation panel code : 

```
    <li [...] aria-label="Cyan Note" style="border-left: 4px solid rgb(190, 211, 243)"    >
        [...]
          <p>Noć je bila vrela, jedna od najtoplijih</p>
          [...]
          <div class="tag">
            <a [...]>This is authored note about the text</a>
          </div>
```

Corresponding JSON keys exported : 

```
       "items":
 [...]
      "body": {
      "value": "This is authored note about the text",
      },
      "target": 
 [...]
        "selector": 
 [...]
            "exact": "Noć je bila vrela, jedna od najtoplijih",
```

**Comment by gautierchomel** [comment]
For now, [Readium annotations](https://github.com/readium/annotations) allow setting a language attribute to [`Body`](https://github.com/readium/annotations/tree/main?tab=readme-ov-file#14-body). We could use the Application Language setting as a default.

I find no clear information about whether it is possible to set up a language for a `selector`; if it is, we could deduce it from the content.

For further works, we can 
* follow up the discussion [Needed text attributes for body:value and TextQuoteSelector #21](https://github.com/readium/annotations/discussions/21)
* think about a possible place in the settings or the annotation editor to set up language for annotations (for all annotations or for this annotation)

**Comment by danielweck** [comment]
The user-entered "note" (for bookmarks and annotations) needs to at least carry `language`, preferably `direction` too (although this can be inferred from the locale BCP47 code, in fact we already do this in other cases).

https://github.com/edrlab/thorium-reader/blob/355e1c96e3cdb0056e450c1595b45ce98112c150/src/common/language-string.ts#L104-L110

https://github.com/edrlab/thorium-reader/blob/355e1c96e3cdb0056e450c1595b45ce98112c150/src/renderer/reader/components/ReaderMenu.tsx#L123-L151

Currently, bookmark and annotation "notes" are rendered by the navigator as small temporary popup floating overlay callouts (that displays on long-ish mouse hover).

https://github.com/edrlab/thorium-reader/blob/355e1c96e3cdb0056e450c1595b45ce98112c150/src/renderer/reader/redux/sagas/note.ts#L90-L94

https://github.com/edrlab/thorium-reader/blob/355e1c96e3cdb0056e450c1595b45ce98112c150/src/renderer/reader/redux/sagas/note.ts#L364-L368

**Comment by danielweck** [comment]
...in other words, `textualValue` itself is not sufficient, we need to somehow enable support for `language` in the GUI and UX (for example as a Spanish user I might wish to type in Arabic inside the "note" text editor).

**Comment by danielweck** [comment]
As for the bookmark / annotation selector: well, the document selection created by users when adding an annotation can span across multiple HTML elements, which may be in different languages, and may contain deep inner text in different languages too. This is easily reproducible with the Wasteland EPUB. For bookmarks, the selector is essentially a point between consecutive characters (at least internally in Thorium), but more generally this would be a single HTML element which is as a document atom can specify or inherit a language, but once again which may contain different languages deep inside (DOM descendants).

**Comment by llemeurfr** [comment]
Before processing this issue, we'll discuss the matter with the developers of citation software (Zotero) and social book networks (Readwise).

**Comment by danielweck** [comment]
Related: https://github.com/edrlab/thorium-reader/issues/2863

**Comment by llemeurfr** [comment]
Language and direction are both required for internationalization.
At the level of an annotation object, two properties are logically impacted: creator and body. The body is much more important (for Screen Reader/TSS). We'll set the creator aside for now. 

The language and direction of the body value are already stored in Thorium, which is good. We now have to surface the language on the note-taking UI. 

This is a proposal, let's discuss it here: 

The language of user annotations is a user property and can be modified in user settings (in a new User section). The default value is the Thorium UI locale; the user will be able to change it via a drop-down or equivalent affordance, and the direction is inferred from the language.  Once set, the lang/dir value is persisted and added to new notes (previously edited notes are not updated). 

On export of an AnnotationSet, if the lang/dir is not equal for all exported annotations, each `body` object is formatted as: 

```json
"body": {
          "type": "TextualBody",
           "value": {
               "language": "he",
                "direction": "rtl",
                "text": "היא שפת סימון."
             }
         },
```

If all exported annotations have an identical lang/dir (which is the major use case), the information is not set at the local level; it is set at the global level of the AnnotationSet (the envelope), as:

```json
"@context": [
              "https://www.w3.org/ns/epub-anno.jsonld",
              {
                 "@language": "he",
                 "@direction": "rtl",

              }
            ],
```

and the `body` objects are simply expressed as:

```json
"body": {
          "type": "TextualBody",
           "value":  "היא שפת סימון."},
```

When imported, such an AnnotationSet is processed so that each Annotation is set with the common lang/dir value. 

There are other strings in natural language: the user's name (creator), the title of an AnnotationSet, the title, authors, and publisher of the book that all annotations are about in a Set. 

I propose that these strings be kept unlocalized in Thorium Reader for the time being, to avoid adding complexity to the internal model for no use in practice (these strings are never localized in EPUB, even if they could be, using `xml:lang`and `dir`).

**Comment by gautierchomel** [comment]
Waiting for the W3C annotation spec to be stable.
